content/post/74-tunneling-http-in-ipfs.md

```diff
@@ -0,0 +1,63 @@
+---
+date: 2019-03-08
+url: 74-tunneling-http-in-ipfs
+tags: libp2p, go-ipfs
+title: Securely tunneling HTTP through IPFS with libp2p
+author: Hector Sanjuan
+---
+
+The last [go-ipfs](https://github.com/ipfs/go-ipfs) release (0.4.19) includes
+[a new experimental feature](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#p2p-http-proxy)
+which allows any peer to serve an HTTP endpoint through the IPFS daemon's
```

Comment:
I can't tell if this is a feature, or scope creeping. It scares the heck out of me the amount of stuff that we glue to IPFS.

Not blocking this (specially because it is already done), that said, would it be up to consideration to make this an external tool before declaring it in a blog post that then makes it harder to change?

Reply:
The powerful thing about this feature is it allows IPFS to trivially blue pill the existing web. By that I mean that any existing web app, regardless of what servers it talks to, can be made to work without significant changes using this proxy.